src/Moq/Protected/ProtectedMock.cs

```diff
@@ -288,6 +288,33 @@ private static void ThrowIfMemberMissing(string memberName, MemberInfo member)
 			}
 		}
 
+		private static void ThrowIfMethodMissing(string methodName, MethodInfo method, object[] args)
+		{
+			if (method == null)
+			{
+				List<string> extractedTypeNames = new List<string>();
+				foreach (object o in args)
+				{
+					if (o is System.Linq.Expressions.MethodCallExpression)
```

Comment:
While I haven't thought about this in great detail, it seems wrong to have such a very specific special case for `o is MethodCallExpression`. The existing method `ProtectedMock.ToArgTypes(object[] args)` suggests that any value in an `args` array might be one of various kinds of `Expression` (when matchers are used), or some constant value (otherwise). Therefore, it would seem right to change this line to `if (o is Expression expr)`.

Reply:
I did wonder about this, cheers for the pointer. You're right, it's probably better to use `Expression`, I hadn't known about it's parent classes and that they'd have a `Type` property too, I've switched it now :)